platform/src/main/java/org/hillview/sketches/results/StringDyadicDecomposition.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2019 VMware Inc. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hillview.sketches.results;
+
+import org.hillview.dataset.api.Pair;
+import org.hillview.table.api.IColumn;
+import org.hillview.table.columns.StringColumnPrivacyMetadata;
+
+import java.util.ArrayList;
+
+public class StringDyadicDecomposition extends DyadicDecomposition<String> {
```

Comment:
"Dyadic" decomposition really just refers to the binary tree decomposition for ordered data, so we shouldn't use that word everywhere, since we want to support trees with curator-defined structure.